(playdate support) delimcc.c is compiled even if Continuations isn't used

To reproduce:

1. Add this line to delimcc.c, just to make it clear that the code is being included
```diff
diff --git a/nativelib/src/main/resources/scala-native/delimcc.c b/nativelib/src/main/resources/scala-native/delimcc.c
index bd00eb6aa..dd0597d1e 100644
--- a/nativelib/src/main/resources/scala-native/delimcc.c
+++ b/nativelib/src/main/resources/scala-native/delimcc.c
@@ -1,4 +1,6 @@
 #if defined(SCALANATIVE_COMPILE_ALWAYS) || defined(__SCALANATIVE_DELIMCC)
+
+#error "this shouldn't happen!"
 #include "delimcc.h"
 #include <stddef.h>
 #include <stdio.h>

```
2. Publish a snapshot
3. Try to link an app/library, even a "hello world" or a no-op main.

Output:

```
[info] Linking (multithreadingEnabled=true, disable if not used) (322 ms)
[info] Discovered 907 classes and 5623 methods after classloading
[info] Checking intermediate code (quick) (4 ms)
[info] Multithreading was not explicitly enabled - initial class loading has not detected any usage of system threads. Multithreading support will be disabled to improve performance.
[info] Linking (multithreadingEnabled=false) (277 ms)
[info] Discovered 774 classes and 4690 methods after classloading
[info] Checking intermediate code (quick) (2 ms)
[info] Discovered 749 classes and 3668 methods after optimization
[info] Optimizing (debug mode) (156 ms)
[info] Produced 43 LLVM IR files
[info] Generating intermediate code (413 ms)
[error] /Users/kubukoz/projects/demos/target/scala-3.3.3/native/dependencies/nativelib_native0.5_3-0/scala-native/delimcc.c:3:2: error: "this shouldn't happen!"
[error] #error "this shouldn't happen!"
[error]  ^
[error] 1 error generated.
[info] Compiling to native code (1524 ms)
[error] stack trace is suppressed; run last Compile / nativeLink for the full output
[error] stack trace is suppressed; run last Compile / nativeLink for the full output
[error] stack trace is suppressed; run last Compile / nativeLink for the full output
[error] stack trace is suppressed; run last Compile / nativeLink for the full output
[error] stack trace is suppressed; run last Compile / nativeLink for the full output
[error] stack trace is suppressed; run last Compile / nativeLink for the full output
[info] Total (2502 ms)
[error] Failed to compile /Users/kubukoz/projects/demos/target/scala-3.3.3/native/dependencies/nativelib_native0.5_3-0/scala-native/delimcc.c
[error] (Compile / nativeLink) Failed to compile /Users/kubukoz/projects/demos/target/scala-3.3.3/native/dependencies/nativelib_native0.5_3-0/scala-native/delimcc.c
```

Example app setup (requires a publish of `0.5.1-SNAPSHOT`): https://github.com/kubukoz/demos/compare/sn-continuations-repro?expand=1

--- a/nativelib/src/main/scala-3/scala/scalanative/runtime/Continuations.scala
+++ b/nativelib/src/main/scala-3/scala/scalanative/runtime/Continuations.scala
@@ -6,7 +6,7 @@ import scala.scalanative.runtime.Intrinsics.*
 import scala.collection.mutable
 
 import scala.util.Try
-import scala.scalanative.meta.LinktimeInfo.isWindows
+import scala.scalanative.meta.LinktimeInfo.isContinuationsSupported
 
 object Continuations:
   import Impl.*
@@ -24,7 +24,8 @@ object Continuations:
    *  We want our implementation to use the allocations done by
    *  `Continuation.alloc`, so that the GC is aware of the stack fragment.
    */
-  Impl.init(CFuncPtr2.fromScalaFunction(allocateBlob))
+  if isContinuationsSupported then
+    Impl.init(CFuncPtr2.fromScalaFunction(allocateBlob))
 
   /** Marks the given body as suspendable with a `BoundaryLabel` that `suspend`
    *  can refer to. Forwards the return value of `body`, or the return value of
@@ -35,8 +36,7 @@ object Continuations:
    *  `boundary` call higher on the same call stack is undefined behaviour.
    */
   inline def boundary[T](inline body: BoundaryLabel[T] ?=> T): T =
-    // Disable on Windows
-    if isWindows then UnsupportedFeature.continuations()
+    if !isContinuationsSupported then UnsupportedFeature.continuations()
 
     val call: ContinuationBody[T] = (x: Impl.BoundaryLabel) =>
       Try(body(using x))

--- a/nativelib/src/main/scala/scala/scalanative/meta/LinktimeInfo.scala
+++ b/nativelib/src/main/scala/scala/scalanative/meta/LinktimeInfo.scala
@@ -45,6 +45,10 @@ object LinktimeInfo {
   )
   def isWeakReferenceSupported: Boolean = resolved
 
+  @resolvedAtLinktime()
+  def isContinuationsSupported: Boolean =
+    isLinux || isMac || isFreeBSD || isOpenBSD || isNetBSD
+
   @resolvedAtLinktime(
     "scala.scalanative.meta.linktimeinfo.isMultithreadingEnabled"
   )

--- a/unit-tests/native/src/test/scala-3/scala/scala/scalanative/runtime/ContinuationsTest.scala
+++ b/unit-tests/native/src/test/scala-3/scala/scala/scalanative/runtime/ContinuationsTest.scala
@@ -4,11 +4,11 @@ import org.junit.Test
 import org.junit.Assert._
 
 import Continuations._
-import scala.scalanative.meta.LinktimeInfo.{isWindows, is32BitPlatform}
+import scala.scalanative.meta.LinktimeInfo.{isContinuationsSupported, is32BitPlatform}
 
 class ContinuationsTest:
   @Test def canBoundaryNoSuspend() =
-    if !isWindows then
+    if isContinuationsSupported then
       val res = boundary[Int] {
         val x = 1
         val y = 2
@@ -17,7 +17,7 @@ class ContinuationsTest:
       assert(res == 3)
 
   @Test def canBoundarySuspend() =
-    if !isWindows then
+    if isContinuationsSupported then
       val res = boundary[Int] {
         val x = 1
         suspend[Int](_ => x + 1)
@@ -26,14 +26,14 @@ class ContinuationsTest:
       assert(res == 2)
 
   @Test def canBoundarySuspendImmediateResume() =
-    if !isWindows then
+    if isContinuationsSupported then
       val r = boundary[Int] {
         1 + suspend[Int, Int](r => r(2)) + suspend[Int, Int](r => r(3)) + 4
       }
       assert(r == 10)
 
   @Test def canBoundarySuspendCommunicate() =
-    if !isWindows then
+    if isContinuationsSupported then
       case class Iter(n: Int, nx: Int => Iter)
       val r0 = boundary[Iter] {
         var r = 0
@@ -49,7 +49,7 @@ class ContinuationsTest:
       assert(r2.n == 5)
 
   @Test def fibonacci(): Unit = {
-    if !isWindows then
+    if isContinuationsSupported then
       import scala.collection.mutable.ArrayBuffer
 
       case class Seqnt[T, R](v: T, nx: R => Seqnt[T, R])
@@ -80,7 +80,7 @@ class ContinuationsTest:
   }
 
   @Test def basic(): Unit = {
-    if !isWindows then
+    if isContinuationsSupported then
       enum Response[T] {
         case Next(nx: () => Response[T], v: T)
         case End(v: T)

--- a/unit-tests/native/src/test/scala-3/scala/scala/scalanative/runtime/ContinuationsTest.scala
+++ b/unit-tests/native/src/test/scala-3/scala/scala/scalanative/runtime/ContinuationsTest.scala
@@ -4,7 +4,10 @@ import org.junit.Test
 import org.junit.Assert._
 
 import Continuations._
-import scala.scalanative.meta.LinktimeInfo.{isContinuationsSupported, is32BitPlatform}
+import scala.scalanative.meta.LinktimeInfo.{
+  isContinuationsSupported,
+  is32BitPlatform
+}
 
 class ContinuationsTest:
   @Test def canBoundaryNoSuspend() =

